internal/errors/ngt.go

```diff
@@ -26,6 +26,13 @@ var (
 
 	ErrIndexNotFound = New("index file not found")
 
+	ErrInvalidDimensionSize = func(current, limit int) error {
+		if limit == 0 {
```

Comment:
is this checking should be 
```suggestion
		if current <=2 {
```
?

Reply:
no, if limit is unset, we want return default error there is no meaning